src/v/cluster/tm_stm.cc

```diff
@@ -117,16 +117,77 @@ tm_stm::tm_stm(
       config::shard_local_cfg().tm_violation_recovery_policy.value())
   , _feature_table(feature_table) {}
 
-ss::future<std::optional<tm_transaction>>
-tm_stm::get_tx(const kafka::transactional_id& tx_id) {
+std::optional<tm_transaction>
+tm_stm::do_get_tx(const kafka::transactional_id& tx_id) {
     auto tx = _mem_txes.find(tx_id);
     if (tx != _mem_txes.end()) {
-        co_return tx->second;
+        return tx->second;
     }
     tx = _log_txes.find(tx_id);
     if (tx != _log_txes.end()) {
-        co_return tx->second;
+        return tx->second;
+    }
+    return std::nullopt;
+}
+
+ss::future<std::optional<tm_transaction>>
+tm_stm::get_tx(const kafka::transactional_id& tx_id) {
```

Comment:
looks like you could take tx_id by value since this is a coroutine and the method seems to block and isn't performance critical?

Reply:
Commented else where.. I think this is perf critical path but I still removed const references as copying is probably trivial.